### ISSUE TITLE
[setters] duplicate

### ISSUE THREAD

**Issue by mesqueeb** [vuex-cloud/rfcs#9]
Should be similar to the way we implement other special setters. We can choose this one last.

some original thoughts:

```javascript
// A. via new action to import (current v-e-f implementation)
dispatch('m/duplicate', id) // or id[]
// resolve into the `newId` or a map: {[oldId]: newId}

// B. via helper fn
dispatch('m/set', duplicate(id, id2))
```

@louisameline 
> Like before, I think we should stick to actions as much as possible, and not force to import functions as well. I'd go for 2.A

**Comment by louisameline** [comment]
I think this should be made in a separate package, outside of the vuex-cloud organisation because:
- people could use that feature without using vuex-cloud-firebase
- vuex-setters should stick to the basic write operations

We should definitely put a link to @mesqueeb/vuex-duplicate on vuex-setters' page though, as well as to others projects in the future that will be "vuex-setters-compatible".
It will be a good example of how external projects can bind on vuex-setters to provide more functionality.

It should hopefully be pretty straightforward, vuex-duplicate will offer a "module/duplicate" action, which will in turn call "module/insert" under the hood, something like that. And voilà, it will work with Firestore sync.

**Comment by louisameline** [comment]
In this duplicate feature, I would add the option to run this in a transaction, or you could end up duplicating a document which is already stale.

**Comment by mesqueeb** [comment]
My personal use case for this is a to-do list app. You can select a todo item & then duplicate the row.

Stale data is something I never struggled with before because the app I had made lists could not be shared among other users and a DBChannel would always be open.

So it's hard for me to imagine a use case where data needs to be checked if it's stale or not. 😄

**Comment by louisameline** [comment]
Yep, but we library makers should think as far beyond our own use cases as possible and make things that work for most people out of the box, avoiding pitfalls for them. Or at least make the caveats very obvious; otherwise I consider the work not finished. Not everyone uses opens realtime streams for example.

**Comment by mesqueeb** [comment]
But it's hard to develop withou understanding the use case. Can you elaborate a bit on an example use case that requires stale checking ? Just so I have an idea.

**Comment by louisameline** [comment]
I don't know, you may want to duplicate a blog post or whatever and delete the original. You want to make sure you duplicate the most up-to-date version before you do. Not everyone uses `openDBChannel`, and even when you do, there's a chance that you don't have the latest version because 1/ you were offline when you gave the order 2/ you were online but a change happened in the few milliseconds after you gave the order.
